crates/ratchet-core/src/ops/cache.rs

```diff
@@ -35,9 +35,9 @@ impl Cache {
         builder: &mut WgslKernelBuilder,
         _: bool,
     ) -> Result<(), OperationError> {
-        builder.register_storage("C", BindingMode::ReadWrite, Array::<P>::default());
-        builder.register_storage("S", BindingMode::ReadOnly, Array::<P>::default());
-        builder.register_storage("D", BindingMode::ReadWrite, Array::<P>::default());
+        builder.register_storage("C", BindingMode::ReadWrite, Array::<vec4<f32>>::default());
```

Comment:
This will only work for `vec4<f32>`!

Reply:
Okay, just learnt that `Vec4<T>` implements `WgslPrimitive` for any T. Making it `Array::<P>` itself, from what I understand.